--- httpd-2.2.27/server/mpm/worker/worker.c 2014-08-07 06:04:52.282532895 -0400
+++ httpd-2.2.27/server/mpm/worker/worker.c 2014-08-07 06:03:07.862946931 -0400
@@ -1254,6 +1254,7 @@ static void child_main(int child_num_arg
          */
         unblock_signal(SIGTERM);
         apr_signal(SIGTERM, dummy_signal_handler);
+        int skip = 0;
         /* Watch for any messages from the parent over the POD */
         while (1) {
             rv = ap_mpm_pod_check(pod);
@@ -1274,6 +1275,8 @@ static void child_main(int child_num_arg
                  */
                 join_start_thread(start_thread_id);
                 signal_threads(rv == AP_GRACEFUL ? ST_GRACEFUL : ST_UNGRACEFUL);
+                /* Hack to avoid deadlock on futex() */
+                skip = 1;
                 break;
             }
         }
@@ -1285,7 +1288,8 @@ static void child_main(int child_num_arg
          *   If the worker hasn't exited, then this blocks until
          *   they have (then cleans up).
          */
-        join_workers(ts->listener, threads);
+        if(!skip)
+          join_workers(ts->listener, threads);
     }

     free(threads);
