--- mod_fcgid-2.3.9/modules/fcgid/mod_fcgid.c 2014-08-14 14:28:45.934038215 +0000
+++ mod_fcgid-2.3.9/modules/fcgid/mod_fcgid.c 2014-08-14 14:26:48.497413899 +0000
@@ -142,6 +142,15 @@ static void fcgid_add_cgi_vars(request_r
 {
     apr_array_header_t *passheaders = get_pass_headers(r);

+    // 2014-03-21 Donatas
+    const char *doc_root = apr_table_get(r->subprocess_env, "BINARY_DOCROOT");
+    const char *server_admin = apr_table_get(r->subprocess_env, "BINARY_SERVERADMIN");
+    if( doc_root != NULL)
+      apr_table_set(r->subprocess_env, "DOCUMENT_ROOT", doc_root);
+    if( server_admin != NULL)
+      apr_table_set(r->subprocess_env, "SERVER_ADMIN", server_admin);
+    // 2014-03-21 END Donatas
+
     if (passheaders != NULL) {
         const char **hdr = (const char **) passheaders->elts;
         int hdrcnt = passheaders->nelts;
@@ -177,6 +186,7 @@ static void fcgid_add_cgi_vars(request_r
             apr_table_setn(e, "SCRIPT_NAME", merge_path);
         }
     }
+
 }

 static int fcgid_handler(request_rec * r)
@@ -287,6 +297,10 @@ static int fcgid_handler(request_rec * r
     ap_add_output_filter_handle(fcgid_filter_handle, NULL, r,
                                 r->connection);

+    /* Hack to keep wrapper from sql, to avoid random changing.. */
+    const char *b_wrapper = apr_table_get(r->notes, "binary-wrapper");
+    if(b_wrapper != NULL && *b_wrapper == '/' && !strncmp(b_wrapper, "/opt/php/wrapper", 16))
+      wrapper_conf->cmdline = b_wrapper;
     http_retcode = bridge_request(r, FCGI_RESPONDER, wrapper_conf);
     return (http_retcode == HTTP_OK ? OK : http_retcode);
 }
